common/config/src/raw.rs

```diff
@@ -15,8 +14,42 @@ pub struct Config {
 }
 
 impl Config {
+    /// Tries to parse from java properties format and then using
     pub fn parse<P: AsRef<Path>>(path: P) -> Result<Self, ConfigError> {
-        Ok(serde_yaml::from_reader(File::open(path)?)?)
+        let is_default_path = path.as_ref().to_string_lossy() == argv::DEFAULT_YAML_FILE;
+        let mut file = match File::open(path) {
```

Comment:
Style nit, I had some difficulty reading this at first glance but that might be because I've spent too long in FP land.

```
let mut file = File::open(path).or_else(|e|
    if !is_default_path {
        // A path was configured but the file was NOT found
        Err(e)
    } else {
        File::open(argv::DEFAULT_CONF_FILE)
    }
)?;
```

Reply:
oh, good suggestion! I'll change it.